src/preflight/handler.js

```diff
@@ -185,22 +186,51 @@ export const preflightAudit = async (context) => {
       pageResult.audits.push({ name: AUDIT_LINKS, type: 'seo', opportunities: [] });
     });
 
-    const { auditResult } = await runInternalLinkChecks(scrapedObjects, context, {
+    const { auditResult } = await runInternalLinkChecks(urls, scrapedObjects, context, {
       pageAuthToken: `token ${pageAuthToken}`,
     });
     if (isNonEmptyArray(auditResult.brokenInternalLinks)) {
-      auditResult.brokenInternalLinks.forEach(({ pageUrl, href, status }) => {
-        const audit = resultMap.get(pageUrl).audits.find((a) => a.name === AUDIT_LINKS);
-        audit.opportunities.push({
-          check: 'broken-internal-links',
-          issue: {
-            url: href,
-            issue: `Status ${status}`,
-            seoImpact: 'High',
-            seoRecommendation: 'Fix or remove broken links to improve user experience and SEO',
-          },
+      if (normalizedStep === AUDIT_STEP_SUGGEST) {
+        const brokenLinks = auditResult.brokenInternalLinks.map((link) => ({
+          urlTo: new URL(site.getBaseURL()).origin + new URL(link.urlTo).pathname.replace(/\/$/, ''),
+          href: link.href,
+          status: link.status,
+        }));
+        log.debug(`[preflight-audit] Found ${JSON.stringify(brokenLinks)} broken internal links`);
+        const brokenInternalLinks = await
+        generateSuggestionData(baseURL, brokenLinks, context, site);
+        log.debug(`[preflight-audit] Generated suggestions for broken internal links: ${JSON.stringify(brokenInternalLinks)}`);
+        brokenInternalLinks.forEach(({
+          urlTo, href, status, urlsSuggested, aiRationale,
+        }) => {
+          const audit = resultMap.get(href).audits.find((a) => a.name === AUDIT_LINKS);
+          const aiUrls = urlsSuggested?.map((url) => (new URL(baseURL).origin + new URL(url).pathname.replace(/\/$/, '')));
+          audit.opportunities.push({
+            check: 'broken-internal-links',
+            issue: {
+              url: urlTo,
+              issue: `Status ${status}`,
+              seoImpact: 'High',
+              seoRecommendation: 'Fix or remove broken links to improve user experience and SEO',
+              urlsSuggested: aiUrls,
+              aiRationale,
+            },
+          });
+        });
+      } else {
+        auditResult.brokenInternalLinks.forEach(({ urlTo, href, status }) => {
+          const audit = resultMap.get(href).audits.find((a) => a.name === AUDIT_LINKS);
```

Comment:
assumes resultMap.get(href) always exists and .audits.find(...) never returns undefined. this can easily throw.